plugins/notion/src/data.ts

```diff
@@ -144,6 +145,20 @@ export async function syncCollection(
                 fieldData[pageContentProperty.id] = { type: "formattedText", value: contentHTML }
             }
 
+            if (fieldsById.has(pageCoverProperty.id)) {
+                let coverValue: string | null = null
+
+                if (item.cover) {
+                    if (item.cover.type === "external") {
+                        coverValue = item.cover.external.url
+                    } else if (item.cover.type === "file") {
+                        coverValue = item.cover.file.url
+                    }
+                }
```

Comment:
nit: for exhaustive check

```suggestion
                if (item.cover) {
                    switch (item.cover.type) {
                        case "external":
                            coverValue = item.cover.external.url
                            break
                        case "file":
                            coverValue = item.cover.file.url
                            break
                        default:
                            item.cover satisfies never
                    }
                }
```